### PR TITLE
Paraview version

### PR DESCRIPTION
**Pull request by danlipsa**

### CODE REVIEW

**Comment by danlipsa** [comment]
Fixes 
https://github.com/spack/spack/pull/12699

**Comment by danlipsa** [comment]
@chuckatkins Please review.

**Comment by adamjstewart** [comment]
Can you keep these:
```python
list_url = "http://www.paraview.org/files"
list_depth = 1
```
This allows `spack versions` and `spack checksum` to find new versions.

**Comment by danlipsa** [comment]
Those don't  seem to make a difference. Adding those lines will still result in the same versions listed for checksum:
```
[~/projects/spack (paraview_version *$<>)]$ spack checksum paraview
==> Found 3 versions of paraview:
  
  5.6.2  https://www.paraview.org/files/v5.6/ParaView-v5.6.2.tar.xz
  5.6.1  https://www.paraview.org/files/v5.6/ParaView-v5.6.1.tar.xz
  5.6.0  https://www.paraview.org/files/v5.6/ParaView-v5.6.0.tar.xz

==> How many would you like to checksum? (default is 1, q to abort) 
```

Is this because I have a url_for_version defined?

**Comment by danlipsa** [comment]
Shouldn't those URLs be used for finding packages for computing checksums?

**Comment by adamjstewart** [comment]
If you replace `http` with `https` if finds a few more versions. I think the reason that it only finds recent versions is that we changed the extension, and older versions don't come as `.tar.gz` files. As long as newer versions come as `.tar.xz` files it is still useful to have.

**Comment by danlipsa** [comment]
Sounds good. Thanks.

File: var/spack/repos/builtin/packages/paraview/package.py [hunk 1]
```diff
@@ -11,14 +11,16 @@ class Paraview(CMakePackage):
...
-    _urlfmt  = 'http://www.paraview.org/files/v{0}/ParaView-v{1}{2}.tar.gz'
+    homepage = 'https://www.paraview.org'
+    url      = "https://www.paraview.org/files/v5.6/ParaView-v5.6.2.tar.xz"
+    list_url = "https://www.paraview.org/files"
+    list_depth = 1
     git      = "https://gitlab.kitware.com/paraview/paraview.git"
 
     maintainers = ['chuckatkins', 'danlipsa']
 
     version('develop', branch='master', submodules=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/paraview/package.py]: You can actually call this version `master` if you want now that #1983 has been merged.

Comment (danlipsa) [reply var/spack/repos/builtin/packages/paraview/package.py]: There is the thought that keeping the version develop will make spack more consistent even if it does not match with the name of the branch. Not sure what is better. Chuck?

Comment (chuckatkins) [reply var/spack/repos/builtin/packages/paraview/package.py]: We could change it but develop has been in for a little bit now and there's still no established spack policy or practice AFAIK so I'd rather leave it as is until such a policy is in place.

**Pull request merged**